src/reactiveData.mli

```diff
@@ -17,66 +17,216 @@
+      inside [d1] and [d2]. (The default value for [eq] is [(=)].) *)
+  val diff : eq:('a -> 'a -> bool) -> 'a data -> 'a data -> 'a patch
+
 end
 
+(** Functor for turning a plain container into an incremental one *)
 module Make(D : DATA) : S with type 'a data = 'a D.data
                            and type 'a patch = 'a D.patch
+
+(** Reactive list data structure *)
 module RList :
 sig
+
+  (** Patch operation on lists. All operations are of linear
+      complexity. *)
   type 'a p =
-      I of int * 'a
-    | R of int
-    | U of int * 'a
-    | X of int * int
+    | I of int * 'a  (** [I i v] adds [v] at position [i] *)
+    | R of int       (** [R i] removes [i]-th element *)
+    | U of int * 'a  (** [U i v] substitutes [i]-th element with [v] *)
+    | X of int * int (** [X i j] swaps the [i]-th and [j]-th elements *)
+
+  (** A patch is a list of patch operations. The operations are
+      applied in the order they appear in the list.
+
+      The indices correspond to list contents after the operations
+      that appear earlier in the list have been applied, not to the
+      contents before the whole patch operation.
+
+      A patch comprised of [I], [R], and [U] steps with increasing
+      indices can be applied in time O(m + n), where m is the patch
+      length and n is the current size of the list. (Arbitrary patches
+      are slower, requiring O(m * n).) *)
+  type 'a patch = 'a p list
+
   include S with type 'a data = 'a list
-             and type 'a patch = 'a p list
+             and type 'a patch := 'a patch
 
+  (** Deprecated. Use [empty]. *)
   val nil : 'a t
+
+  (** Add element to the end *)
   val append : 'a -> 'a handle -> unit
+
+  (** Add element to the beginning *)
   val cons : 'a -> 'a handle -> unit
+
+  (** [insert v i h] adds [v] as the [i]-th position in the container
+      corresponding to [h]. The indices of the subsequent elements
+      change. *)
   val insert : 'a -> int -> 'a handle -> unit
+
+  (** [remove i h] removes the [i]-th position from the container
+      corresponding to [h]. The indices of the subsequent elements
+      change. *)
   val remove : int -> 'a handle -> unit
+
+  (** [update v i h] substitutes the [i]-th element of the container
+      corresponding to [h] with [v] *)
   val update : 'a -> int -> 'a handle -> unit
+
+  (** [move i j h] moves the [i]-th element of the container
+      corresponding to [h] to the [j]-th position, modifying the
+      indices of other elements *)
   val move : int -> int -> 'a handle -> unit
 
+  (** Produce container list containing a single, constant element *)
   val singleton : 'a -> 'a t
+
+  (** Produce reactive list containing a single element that gets
+      updated based on a signal *)
   val singleton_s : 'a React.S.t -> 'a t
+
+  (** [concat a b] is the concatenation of [a] and [b], and it gets
+      updated whenever [a] and [b] change *)
   val concat : 'a t -> 'a t -> 'a t
```

Comment:
Ideally, this should be called `append`, but this one is a bit trickier (because there was an `append` before).